test/formats/jsonFlat.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+var assert = require('chai').assert,
+  formats = require('../../lib/common/formats');
+
+var formatter = formats['json/flat'];
+
+describe('formats', function () {
+  describe('json/flat should', function () {
+    it('evaluate complex object with property referencing another value', function () {
+      var dictionary = {
```

Comment:
Take a look at test/formats/scssVariables.js for a better example how to test this formatter.